packages/google_maps_flutter/google_maps_flutter_web/lib/src/markers.dart

```diff
@@ -61,15 +63,23 @@ abstract class MarkersController<T extends JSObject, O>
     final MarkerController<T, O>? markerController =
         _markerIdToController[marker.markerId];
     final T? currentMarker = markerController?.marker;
+    final O markerOptions =
+        await _markerOptionsFromMarker(marker, currentMarker);
     final MarkerController<T, O> controller =
-        await createMarkerController(marker, currentMarker, gmInfoWindow);
+        await createMarkerController(marker, markerOptions, gmInfoWindow);
     _markerIdToController[marker.markerId] = controller;
   }
 
-  /// Creates a [MarkerController] for a [Marker] object.
+  /// Creates a [MarkerController] for the given [marker].
+  ///
+  /// [markerOptions] contains configuration that should be used to create
+  /// a [gmaps.Marker] or [gmaps.AdvancedMarkerElement] object. [markersOptions]
+  /// is either [gmaps.MarkerOptions] or [gmaps.AdvancedMarkerElementOptions].
+  ///
+  /// [gmInfoWindow] is marker's info window to show on tap.
   Future<MarkerController<T, O>> createMarkerController(
     Marker marker,
-    T? currentMarker,
+    O markerOptions,
     gmaps.InfoWindow? gmInfoWindow,
   );
```

Comment:
Thanks for this change, I think this makes more sense now (to me, at least! :P)

Reply:
No problem, I think it's better this way, too 👍